### ISSUE TITLE
why is the object of loss class callable?

### ISSUE THREAD

**Issue by ardeal** [pytorch/pytorch#56971]
Hi,

as I know, if the object of the class python is callable,  <code>\_\_call\_\_</code> function should be implemented in the class. however, in loss class of pytorch, I did see the implementation of <code>\_\_call\_\_</code>

for example:
```python
lossfunc = torch.nn.SmoothL1Loss()
loss = lossfunc(p,y)

```
